VS17.0 build error

## Expected behavior and actual behavior.

Error during build `frmts/gtiff/tif_jxl.c`:
```
frmts/gtiff/tif_jxl.c(295): error C2059: syntax error: '}'
frmts/gtiff/tif_jxl.c(487): error C2059: syntax error: '}'
frmts/gtiff/tif_jxl.c(492): error C2059: syntax error: '}'
frmts/gtiff/tif_jxl.c(577): error C2059: syntax error: '}'
```

## Operating system

win11

## GDAL version and provenance

3.4.0; custom build system

---

Replacing `= {};` with `= {0};` in these places fixes the issue.

--- a/frmts/gtiff/tif_jxl.c
+++ b/frmts/gtiff/tif_jxl.c
@@ -292,7 +292,7 @@ JXLPreDecode(TIFF* tif, uint16_t s)
             return 0;
         }
 
-        JxlPixelFormat format = {};
+        JxlPixelFormat format = {0};
         format.num_channels = td->td_planarconfig == PLANARCONFIG_CONTIG ?
                                                     td->td_samplesperpixel : 1;
         format.data_type = jxlDataType;
@@ -484,12 +484,12 @@ JXLPostEncode(TIFF* tif)
             return 0;
         }
 
-        JxlPixelFormat format = {};
+        JxlPixelFormat format = {0};
         format.data_type = GetJXLDataType(tif);
         format.endianness = JXL_NATIVE_ENDIAN;
         format.align = 0;
 
-        JxlBasicInfo basic_info = {};
+        JxlBasicInfo basic_info = {0};
         JxlEncoderInitBasicInfo(&basic_info);
         basic_info.xsize = sp->segment_width;
         basic_info.ysize = sp->segment_height;
@@ -574,7 +574,7 @@ JXLPostEncode(TIFF* tif)
             return 0;
         }
 
-        JxlColorEncoding color_encoding = {};
+        JxlColorEncoding color_encoding = {0};
         JxlColorEncodingSetToSRGB(&color_encoding, /*is_gray*/
             (td->td_planarconfig==PLANARCONFIG_SEPARATE ||
             td->td_samplesperpixel <= 2));

